lib/perl/Genome/InstrumentData/Command/Import/GenerateBase.pm

```diff
@@ -0,0 +1,180 @@
+package Genome::InstrumentData::Command::Import::GenerateBase;
+
+use strict;
+use warnings;
+
+use Genome;
+
+require File::Basename;
+require List::MoreUtils;
+use Params::Validate qw( :types );
+use Text::CSV;
+
+class Genome::InstrumentData::Command::Import::GenerateBase {
+    is => 'Command::V2',
+    is_abstract => 1,
+    has_input => {
+        file => {
+            is => 'Text',
+            doc => 'Comma (.csv) or tab (.tsv) separated file of entity names, attributes and other meta data. Separartor is determined by file extension.',
+        },
+    },
+    has_optional_output => {
+        output_file => {
+            is => 'Text',
+            default_value => '-',
+            doc => 'Output file to put the commands to create the needed entities for import.',
+        },
+    },
+    has_optional_transient => {
+        _output => { is => 'ARRAY', default_value => [], },
+        _instdata_property_names => { is => 'ARRAY', },
+        _entity_attributes => { is => 'ARRAY', },
+        _nomenclature => { is => 'Text', },
+        _names_seen => { is => 'HASH', default_value => {}, },
+    }
+};
+
+sub help_detail {
+    return <<HELP;
+The file should be a comma or tab separated values and indicated with the appropriate extension (csv and tsv). Column headers to use to generate the create commands should start with the entity (individual, sample, library, instdata) name then a period (.) and then then attribute name (Ex: sample.name_part). Here are some required and optional columns. For more, see each entity's create command (Ex: genome sample create --h). Please see Confluence documentation for more information and a full example.
+
```

Comment:
The wiki has a real life example from an SRA metadata file. I guess the _s means string? I need to add the finished spreadsheet to the wiki. Did I answer your Q?